### PR TITLE
[4.2] Don't build tarball update packages

### PR DESCRIPTION
**Pull request by brianteeman**
Pull Request for Issue #39488 

### Summary of Changes
Stops building unusable update packages in non-zip file formats. Neither the joomla upgrade component or the joomla cli update use them and the manual upload and update process in the oomla upgrade component will not accept them (but you have to wait until you have uploaded the file first to find that out).

### Testing Instructions
code review and run the build.php script

### Actual result BEFORE applying this Pull Request
full and update packages created for non zip file formats

### Expected result AFTER applying this Pull Request
no change in the available full packages but onyl the zip format update package is generated

PR has been made against 4.2 so that the packages for 4.3 are generated correctly

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: build/build.php [hunk 1]
```diff
@@ -36,10 +36,10 @@ function usage(string $command)
...
-    echo PHP_TAB . PHP_TAB . '--exclude-gzip:' . PHP_TAB . PHP_TAB . PHP_TAB . 'Exclude the generation of .tar.gz packages' . PHP_EOL;
-    echo PHP_TAB . PHP_TAB . '--exclude-bzip2:' . PHP_TAB . PHP_TAB . 'Exclude the generation of .tar.bz2 packages' . PHP_EOL;
-    echo PHP_TAB . PHP_TAB . '--include-zstd:' . PHP_TAB . PHP_TAB . PHP_TAB . 'Include the generation of .tar.zst packages' . PHP_EOL;
+    echo PHP_TAB . PHP_TAB . '--exclude-zip:' . PHP_TAB . PHP_TAB . PHP_TAB . 'Exclude the generation of full, update and patch zip packages' . PHP_EOL;
+    echo PHP_TAB . PHP_TAB . '--exclude-gzip:' . PHP_TAB . PHP_TAB . PHP_TAB . 'Exclude the generation of full .tar.gz packages. (Update and patch .tar.gz packages are never created.)' . PHP_EOL;
+    echo PHP_TAB . PHP_TAB . '--exclude-bzip2:' . PHP_TAB . PHP_TAB . 'Exclude the generation of full .tar.bz2 packages. (Update and patch .tar.bz2 packages are never created.)' . PHP_EOL;
+    echo PHP_TAB . PHP_TAB . '--include-zstd:' . PHP_TAB . PHP_TAB . PHP_TAB . 'Include  the generation of full .tar.zstd packages. (Update and patch .tar.zstd packages are never created.)' . PHP_EOL;
```
Comment (richard67) [inline build/build.php]:
```suggestion
    echo PHP_TAB . PHP_TAB . '--include-zstd:' . PHP_TAB . PHP_TAB . PHP_TAB . 'Include the generation of full .tar.zstd packages. (Update and patch .tar.zstd packages are never created.)' . PHP_EOL;
```

Comment (brianteeman) [reply build/build.php]: remove extra space

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 00f461263c80d031df1b81c01907371c178569b5<br /><br />I've run the build script with diverse combinations of call parameters, and all worked as expected.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39567">issues.joomla.org/tracker/joomla-cms/39567</a>.</sub>

**Comment by richard67** [comment]
@brianteeman Could you complete the documentation check boxes in the description? On docs.joomla.org it will not need any change because as far as I know there is no documentation about the build script. With manual.joomla.org I'm not sure. Maybe @HLeithner knows if it needs some change there?

**Comment by brianteeman** [comment]
I dont think it needs docs on either of those sites - its an internal task just in a public repo.

But happy write if needed

**Comment by HLeithner** [comment]
Please don't remove the code, instead disable the default build step.

**Comment by richard67** [comment]
> Please don't remove the code, instead disable the default build step.

@HLeithner Does it really need that? If we one day can make the updater use these formats, the code for creating these packages can be easily copied from the code for the full installation packages here https://github.com/joomla/joomla-cms/blob/00f461263c80d031df1b81c01907371c178569b5/build/build.php#L553-L584 and then be modified.

It would complicate our logic unnecessarily if we want to keep that just for later use and disable ny additional logic for the call parameters.

**Comment by wojtekxtx** [comment]
> Does it really need that?

@richard67: Yes, it does.

> the code for creating these packages can be easily copied from the code for the full installation packages here

@richard67 when you copy code you:
* make J! harder to maintain (you have to maintain 2 separate, but identical, copies.
* have to document it fully,
* introduce yet another `PAV`. Its undesirable from security standpoint.
* introduce confusion amongst devs which copy is "the right one" and which copy to `extend` classes against.
* many more things.

**Review by wojtekxtx — COMMENTED** [review]
Simply NO-GO

**Comment by wojtekxtx** [comment]
@richard67 Im offending noone, I just voice my opinions. You are free to disagree with me, Im taking no grudge against you, aas one can disagree on sth, normal thing. If you disagree than you can voice your opinions as well.

But, please, do not imply that my comments are silly; I may not have the level of knowledge insider like you have, but than again, just explain why Im wrong. Its a lot nicer this way.

**Comment by brianteeman** [comment]
Ignoring the troll

@HLeithner What is your logic for keeping something that will never be used. Usually you object to having extra things as it means more to maintain etc. Ju8st trying to understand

**Comment by brianteeman** [comment]
@wojtekxtx if you had a clue what you were talking about you would see that this PR removes code that is not used. But you wanting to keep it means it does all the things you say are bad.

**Comment by wojtekxtx** [comment]
@brianteeman Seriously? Dont you see that more features = more = code = harder to maintain? Its plain simple man.

**Comment by richard67** [comment]
> @brianteeman Seriously? Dont you see that more features = more = code = harder to maintain? Its plain simple man.

@wojtekxtx Where does this PR add new features and new code? It removes unused code, as already said. Your statement again proves that you comment without having checked what this PR here does.

**Comment by carlitorweb** [comment]
> I've run the build script with diverse combinations of call parameters

@richard67 can you tell me how do this, for run the test I will do here?

**Comment by brianteeman** [comment]
@richard67 they've already had two strikes

**Comment by richard67** [comment]
> > I've run the build script with diverse combinations of call parameters
> 
> @richard67 can you tell me how do this? for run the test I will do here...

It needs to have a git clone. Then you can fetch the changes from this PR into a branch and check out that branch on the command line:

```
git fetch upstream pull/39567/head:test-pr-39567
git checkout test-pr-39567
```
Then, still on the command line with current directory = the Joomla root, you can run `php ./build/build.php --help` to get a help about the parameters.

For making a build based on that branch, use the `--remote=HEAD` parameter.

E.g. `php ./build/build.php --remote=HEAD --exclude-gzip --exclude-bzip2` and so on.

**_Important:_** It either needs to run on Linux or another unixoid OS, or when on Windows it needs WSL (Windows Subsystem for Linux) for running the build script because it uses unixoid system commands.

**Comment by carlitorweb** [comment]
> It needs to have a git clone. Then you can fetch the changes from this PR into a branch and check out that branch on the command line:

Yes this one already have it, thank.

> It either needs to run on Linux or another unixoid OS, or when on Windows it needs WSL (Windows Subsystem for Linux) for running the build script because it uses unixoid system commands.

Copy that

**Comment by carlitorweb** [comment]
@brianteeman @richard67  I apply the patch and ran: `php ./build/build.php --remote=4.2-dev` but the tar package was builded. is how should be or I not doing the things right?
build

**Comment by richard67** [comment]
@carlitorweb The tar packages in your screenshot are the full installation packages. They are still built (if not excluded by the parameter). What this PR here does is stop the update packages being built in that format.

**Comment by carlitorweb** [comment]
@richard67 I was sure I was missing something. Thank, I will use others paramaters before end the test. Thank again

**Comment by richard67** [comment]
@carlitorweb Here the tests which I have made, as an inspiration for other testers.

1. php ./build/build.php --help
Result: The changes help texts are shown and they are correct.
```
Usage: php ./build/build.php [options]
	[options]:
		--remote=<remote>:		The git remote reference to build from (ex: `tags/3.8.6`, `4.0-dev`), defaults to the most recent tag for the repository
		--exclude-zip:			Exclude the generation of full, update and patch zip packages
		--exclude-gzip:			Exclude the generation of full .tar.gz packages. (Update and patch .tar.gz packages are never created.)
		--exclude-bzip2:		Exclude the generation of full .tar.bz2 packages. (Update and patch .tar.bz2 packages are never created.)
		--include-zstd:			Include  the generation of full .tar.zstd packages. (Update and patch .tar.zstd packages are never created.)
		--disable-patch-packages:	Disable the generation of patch packages
		--help:				Show this help output
```

2. php ./build/build.php --remote=HEAD
Result: In folder "./build/tmp/packages"
- Joomla_4.2.6_to_4.2.7-dev-Development-Patch_Package.zip
- Joomla_4.2.7-dev-Development-Full_Package.tar.bz2
- Joomla_4.2.7-dev-Development-Full_Package.tar.gz
- Joomla_4.2.7-dev-Development-Full_Package.zip
- Joomla_4.2.7-dev-Development-Update_Package.zip
- Joomla_4.2.x_to_4.2.7-dev-Development-Patch_Package.zip

3. php ./build/build.php --remote=HEAD --exclude-gzip
Result: In folder "./build/tmp/packages"
- Joomla_4.2.6_to_4.2.7-dev-Development-Patch_Package.zip
- Joomla_4.2.7-dev-Development-Full_Package.tar.bz2
- Joomla_4.2.7-dev-Development-Full_Package.zip
- Joomla_4.2.7-dev-Development-Update_Package.zip
- Joomla_4.2.x_to_4.2.7-dev-Development-Patch_Package.zip

4. php ./build/build.php --remote=HEAD --exclude-bzip2
Result: In folder "./build/tmp/packages"
- Joomla_4.2.6_to_4.2.7-dev-Development-Patch_Package.zip
- Joomla_4.2.7-dev-Development-Full_Package.tar.gz
- Joomla_4.2.7-dev-Development-Full_Package.zip
- Joomla_4.2.7-dev-Development-Update_Package.zip
- Joomla_4.2.x_to_4.2.7-dev-Development-Patch_Package.zip

5. php ./build/build.php --remote=HEAD --exclude-gzip --exclude-bzip2
Result: In folder "./build/tmp/packages"
- Joomla_4.2.6_to_4.2.7-dev-Development-Patch_Package.zip
- Joomla_4.2.7-dev-Development-Full_Package.zip
- Joomla_4.2.7-dev-Development-Update_Package.zip
- Joomla_4.2.x_to_4.2.7-dev-Development-Patch_Package.zip

6. php ./build/build.php --remote=HEAD --exclude-gzip --exclude-bzip2 --include-zstd
Result: In folder "./build/tmp/packages"
- Joomla_4.2.6_to_4.2.7-dev-Development-Patch_Package.zip
- Joomla_4.2.7-dev-Development-Full_Package.tar.zst
- Joomla_4.2.7-dev-Development-Full_Package.zip
- Joomla_4.2.7-dev-Development-Update_Package.zip
- Joomla_4.2.x_to_4.2.7-dev-Development-Patch_Package.zip

7. php ./build/build.php --remote=HEAD --disable-patch-packages
Result: In folder "./build/tmp/packages"
- Joomla_4.2.7-dev-Development-Full_Package.tar.bz2
- Joomla_4.2.7-dev-Development-Full_Package.tar.gz
- Joomla_4.2.7-dev-Development-Full_Package.zip
- Joomla_4.2.7-dev-Development-Update_Package.zip

8. php ./build/build.php --disable-patch-packages
Result: In folder "./build/tmp/packages"
- Joomla_4.3.0-alpha2-Alpha-Full_Package.tar.bz2
- Joomla_4.3.0-alpha2-Alpha-Full_Package.tar.gz
- Joomla_4.3.0-alpha2-Alpha-Full_Package.zip
- Joomla_4.3.0-alpha2-Alpha-Update_Package.zip
Result: In folder "./build/tmp"
- checksums.txt
- github_release.txt

**Comment by richard67** [comment]
P.S.: Without the PR, tar.gz and tar.bz2 packages are also built for update and patch packages if not excluded by parameter.

**Comment by carlitorweb** [comment]
@richard67 thank for the explanation

**Comment by carlitorweb** [comment]
I have tested this item :white_check_mark: successfully on 00f461263c80d031df1b81c01907371c178569b5<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39567">issues.joomla.org/tracker/joomla-cms/39567</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39567">issues.joomla.org/tracker/joomla-cms/39567</a>.</sub>

**Comment by laoneo** [comment]
This should go into 5.0 and not in a patch release. I know it is a fix, but we do not know what is done with these tarballs into the wild, so users will expect that it works during a major series.

**Comment by wojtekxtx** [comment]
@richard67 Seriously I do not follow your thinking way......

**Comment by brianteeman** [comment]
> Please don't remove the code, instead disable the default build step.

@hleithner Could you explain the reasoning behind this. Trying to understand the logic of keeping and maintaining code that has no use.



> This should go into 5.0 and not in a patch release. I know it is a fix, but we do not know what is done with these tarballs into the wild, so users will expect that it works during a major series.

I can see some semblance of logic for it not going in a patch release but not one for a minor.

As I wrote in the PR description.

> PR has been made against 4.2 so that the packages for 4.3 are generated correctly

There is no use case for the tarballs. They cannot be used by the joomla update.

And for once we have actual statistics that we can rely on.

tar.gz 0.0014%
tar.bz2 0.0001%

**Comment by wojtekxtx** [comment]
> Could you explain the reasoning behind this?

@brianteeman Sure thing. I **think** what @HLeithner has in mind is tht this code can be reused later-on (even if its of no use now). Its reasonabkle way of thinking.

> tar.gz 0.0014%
tar.bz2 0.0001%

Where are these stats taken from @brianteeman ?

**Comment by roland-d** [comment]
I am in favor of this PR but agree with @laoneo that a bugfix release is not the right release. Who knows what these users do with them, but breaking whose ever flow this is part of is not the way to go.

This needs to go into 5.0.

> Where are these stats taken from @brianteeman ?

If you ask me, just take the amount of downloads divide by the total amount of downloads and times 100%. At the time of writing this is `248/(164035+248+194)*100 = 0.15%`

**Comment by brianteeman** [comment]
sorry my bad math - forgot the * 100

**Comment by brianteeman** [comment]
I guess its a matter of impact.

**Comment by wojtekxtx** [comment]
> impact.

@brianteeman dare to explain?

**Comment by Hackwar** [comment]
I do support this PR. We shouldn't build stuff we rarely use and which just wastes space and cpu cycles... However, while the impact right now is small, there is an impact and thus I also would change this to Joomla 5.0. I'm going to switch over the branch.

**Comment by Hackwar** [comment]
Scrap that, I'm not going to fix those conflicts. :-D

**Comment by brianteeman** [comment]
I still believe that the impact is non-existant as the tarball cannot be used and any downloads we have recorded are the exact thing we want to avoid - downloading something that wont work.

I do not believe that the provision of files in a specified format could ever be part of a B/C promise

**Pull request closed without merging**